### PR TITLE
Feature/stateful alert

### LINKED ISSUE #27: [Alert] Add z-index property on Alert component
Add an optional prop to set the component's `z-index` css prop.

### LINKED ISSUE #28: [Alert] Implement a stateful version of the component with awareness of its own visibility
The current Alert component is a steteless "dumb" functional component and the responsibility of displaying it and hiding it is delegated to higher order components.

It may be useful to include in the package a version of the component that is able to hide itself independently of other components, either with a new property ``visibleTime`` or with a default onDismiss function that hides the Alert.

### PR DESCRIPTION
**Pull request by fczuardi**
- [x] closes #28 
- [x] closes #27

### CODE REVIEW

File: packages/alert/src/BrieflyDisplay.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+//@flow
+import * as React from "react";
+import PropTypes from "prop-types";
+
+const DEFAULT_TIME = 2000;
+
+class BrieflyDisplay extends React.Component<Object, Object> {
+  static propTypes = {
+    time: PropTypes.number,
+    children: PropTypes.node
```
Comment (alcferreira) [inline packages/alert/src/BrieflyDisplay.js]: set `children` as required

File: packages/alert/src/Alert.js [hunk 2]
```diff
@@ -3,40 +3,63 @@ import * as React from "react";
...
+    if (!this.state.isVisible) {
+      return null;
+    }
+    const { visibleTime, zIndex } = this.props;
+    const dismissHandler = () => {
+      this.setState({ isVisible: false });
+      this.props.onDismiss();
+    };
+    const alert = (
+      <StyledAlert type={this.props.type} {...(zIndex ? { zIndex } : {})}>
```
Comment (alcferreira) [inline packages/alert/src/Alert.js]: Hmmm can we pass zIndex like this?

`<StyledAlert type={this.props.type} zIndex={this.props.zIndex}>`

File: packages/alert/src/Alert.js [hunk 3]
```diff
@@ -3,40 +3,63 @@ import * as React from "react";
...
+              onClick={dismissHandler}
+            />
+          </div>
+        )}
+      </StyledAlert>
+    );
+    if (!visibleTime) {
+      return alert;
+    }
+    return <BrieflyDisplay time={visibleTime}>{alert}</BrieflyDisplay>;
```
Comment (alcferreira) [inline packages/alert/src/Alert.js]: Nice! :ok_hand:

File: packages/alert/src/Alert.js [hunk 4]
```diff
@@ -3,40 +3,63 @@ import * as React from "react";
...
-  type: AlertTypes.NEWS,
-  dismissable: true,
-  onDismiss: () => null
-};
+  render() {
+    if (!this.state.isVisible) {
+      return null;
+    }
+    const { visibleTime, zIndex } = this.props;
+    const dismissHandler = () => {
```
Comment (alcferreira) [inline packages/alert/src/Alert.js]: Move this function declaration outside the render function

File: packages/alert/src/Alert.js (hunk 2 above)
Comment (fczuardi) [reply packages/alert/src/Alert.js]: it may require a small change on the styled component, but sure

**Comment by henlz** [comment]
I'm having some troubles to run storybook. Apparently the modules `@crave/farmblocks-...`are not "installed".

Shouldn't this be installed/referenced somewhere?

screen shot 2017-10-26 at 17 01 40

**Comment by fczuardi** [comment]
@henlz re: https://github.com/CraveFood/farmblocks/pull/67#issuecomment-339766916

The packages are linked by the yarn workspaces, if you do these 3 steps:
``yarn install``, ``yarn build`` and ``yarn storybook`` you should end up with a storybook dev server that knows where each dev package lives. Please let me know if this fixes.

**Comment by fczuardi** [comment]
I need to tweak some css a little more, just a sec.

**Pull request merged**